rust/connlib/libs/tunnel/src/tun_android.rs

```diff
@@ -28,30 +29,33 @@ impl Drop for IfaceDevice {
 }
 
 impl IfaceDevice {
-    fn write(&self, _buf: &[u8]) -> usize {
-        tracing::error!("`write` unimplemented on Android");
-        0
+    fn write(&self, buf: &[u8]) -> usize {
+        match unsafe { write(self.fd, buf.as_ptr() as _, buf.len() as _) } {
+            -1 => 0,
+            n => n as usize,
+        }
     }
 
-    pub async fn new() -> Result<Self> {
-        // TODO: This won't actually work for non-root users...
-        let fd = unsafe { open(b"/dev/net/tun\0".as_ptr() as _, O_RDWR) };
-        // TODO: everything!
-        if fd == -1 {
-            Err(Error::Io(std::io::Error::last_os_error()))
-        } else {
-            Ok(Self { fd })
-        }
+    pub async fn new(fd: Option<i32>) -> Result<Self> {
```

Comment:
This is actually unsafe since `fd` might not be valid and we assume it's vaild in the `read`/`write` implementations. Since we are going to change how we obtain the `fd` it's not that important right now but we need to remember that for the refactor.

Reply:
Added

Reply:
Aaaand removed so this will compile